fili-core/src/main/java/com/yahoo/bard/webservice/data/config/table/BaseTableLoader.java

```diff
@@ -159,6 +155,113 @@ public void loadLogicalTableWithGranularities(
     }
 
     /**
+     * Load a new physical table into the dictionary and return the loaded physical table.
+     *
+     * @param definitions  A config object for the physical table
+     * @param dictionaries  The resource dictionaries for reading and storing resource data
+     *
+     * @return The physical table created
+     */
+    protected LinkedHashSet<PhysicalTable> loadPhysicalTablesWithDependency(
+            Set<PhysicalTableDefinition> definitions,
+            ResourceDictionaries dictionaries
+    ) {
+        LinkedHashSet<PhysicalTable> loadedTables = new LinkedHashSet<>();
+
+        Map<TableName, PhysicalTableDefinition> mutableCopyDefinitionMap =
+                buildPhysicalTableDefinitionDictionary(definitions);
+
+        definitions.forEach(definition -> buildPhysicalTablesWithDependency(
```

Comment:
This construct feels odd... It's like we're doing a recursion, but modifying a structure and filling up a bucket as we go. I think things would be much cleaner if we used a more traditional DepthFirstBuilding approach, where the return value of the method returns a LinkedHashSet of all of the tables it built, or perhaps just the 1 table it wanted to build.